test/jdk/java/nio/Buffer/CharBufferAsCharSequenceTest.java

```diff
@@ -0,0 +1,332 @@
+                args.add(Arguments.of(behbbAsCB, buf, i, buf.length - i, "BE HeapByteBuffer as CharBuffer index " + i + " to " + end));
+
+                CharBuffer ledbbAsCB = ByteBuffer.allocateDirect(buf.length * 2)
+                                                 .order(ByteOrder.LITTLE_ENDIAN)
+                                                 .asCharBuffer()
+                                                 .put(buf)
+                                                 .position(1)
+                                                 .slice()
+                                                 .position(i - 1)
+                                                 .limit(end - 1);
+                args.add(Arguments.of(ledbbAsCB, buf, i, buf.length - i, "LE DirectByteBuffer as CharBuffer index " + i + " to " + end));
+            }
+        }
+        return args;
+    }
+
+    @ParameterizedTest(name="{4}")
+    @MethodSource("charBufferArguments")
+    void testToString(CharSequence actual, char[] expected, int start, int stop, String description) {
+        assertEquals(new String(expected, start, stop - start), actual.toString(), description);
+    }
+
+    @ParameterizedTest(name="{4}")
+    @MethodSource("charBufferArguments")
+    void testLength(CharSequence actual, char[] expected, int start, int stop, String description) {
+        assertEquals(stop - start, actual.length(), description);
+    }
+
+    @ParameterizedTest(name="{4}")
+    @MethodSource("charBufferArguments")
+    void testGetCharsRange(CharSequence actual, char[] expected, int start, int stop, String description) {
+        char[] val = new char[16];
+        int length = Math.min(10, stop - start - 1);
+        actual.getChars(1, length + 1, val, 3);
+
+        for (int i = 0; i < length; ++i) {
+            assertEquals(expected[i + start + 1], val[i + 3], "val at offset of " + i + " from " + description);
+        }
+        // test that calling getChars did not move the position
+        assertEquals(expected[start], actual.charAt(0), "first char after calling getChars: " + description);
+    }
+
+    @ParameterizedTest(name="{4}")
+    @MethodSource("charBufferArguments")
+    void testGetCharsAll(CharSequence actual, char[] expected, int start, int stop, String description) {
+        char[] val = new char[stop - start];
+        actual.getChars(0, val.length, val, 0);
+
+        for (int i = 0; i < val.length; ++i) {
+            assertEquals(expected[i + start], val[i], "val at offset of " + i + " from " + description);
+        }
+        // test that calling getChars did not move the position
+        assertEquals(expected[start], actual.charAt(0), "first char after calling getChars: " + description);
+    }
+
+    @ParameterizedTest(name="{4}")
+    @MethodSource("charBufferArguments")
+    void testGetCharsNegativeSourceBeg(CharSequence actual, char[] expected, int start, int stop, String description) {
```

Comment:
The parameter that is being tested here is "srcBegin" so maybe using that naming instead of switching to "SourceBeg" (same comment for several other methods).